src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -86,8 +86,28 @@ public async Task<(JObject containerObject, ArraySegment<string> remaining)> Res
             if (methodInfo == null)
                 return (null, null);
 
+            string[] parts;
+            if (includeFullName)
+            {
+                string fullName = methodInfo.IsAsync == 0 ? methodInfo.TypeInfo?.FullName : StripAsyncPartOfFullName(methodInfo.TypeInfo?.FullName);
+                string[] fullNameParts = fullName.Split(".", StringSplitOptions.TrimEntries);
+                int overlappingPartIdx = Enumerable.Range(0, fullNameParts.Length).LastOrDefault(i => fullNameParts[i] == expressionParts[0]);
+                if (overlappingPartIdx == 0)
```

Comment:
Wouldn't it return `0`, if `fullNameParts[0]` matched OR if nothing matched?

Reply:
Yes, if it matched for the 0th index then we can take the whole `fullNameParts` into consiferation. If there's no match it's the same story - we are interested in the whole `fullNameParts` to be our prefix. I thought that if I write 
```
// default val was returned for overlappingPartIdx
```
it would be understandable but maybe I'll change for: 
```
// either whole expression overlap or nothing matched and default value was returned
```